Makefile

```diff
@@ -139,24 +147,37 @@ OBJS += \
 	pow_fpga_accel.o
 endif
 
+ifeq ("$(BUILD_REMOTE)", "1")
+OBJS += \
+	remote_common.o \
+	remote_interface.o
+
+WORKER_OBJS := $(addprefix $(OUT)/remote-,$(filter-out remote_interface.o, $(OBJS)))
+WORKER_CFLAGS := $(filter-out -DENABLE_REMOTE, $(CFLAGS))
+endif
+
 OBJS := $(addprefix $(OUT)/, $(OBJS))
 
 $(OUT)/test-%.o: tests/test-%.c $(LIBTUV_PATH)/include
 	$(VECHO) "  CC\t$@\n"
 	$(Q)$(CC) -o $@ $(CFLAGS) -I $(SRC) $(LIBTUV_INCLUDE) -c -MMD -MF $@.d $<
 
-$(OUT)/%.o: $(SRC)/%.c $(LIBTUV_PATH)/include
+$(OUT)/%.o: $(SRC)/%.c $(LIBTUV_PATH)/include $(LIBRABBITMQ_PATH)/build/include
 	$(VECHO) "  CC\t$@\n"
-	$(Q)$(CC) -o $@ $(CFLAGS) $(LIBTUV_INCLUDE) -c -MMD -MF $@.d $<
+	$(Q)$(CC) -o $@ $(CFLAGS) $(LIBTUV_INCLUDE) $(LIBRABBITMQ_INCLUDE) -c -MMD -MF $@.d $<
 
-$(OUT)/test-%: $(OUT)/test-%.o $(OBJS) $(LIBTUV_LIBRARY)
+$(OUT)/test-%: $(OUT)/test-%.o $(OBJS) $(LIBS) $(LIBTUV_LIBRARY)
```

Comment:
I was thinking about the consistency.
The `rabbitmq` library is added into the `$(LIBS)` but `libtuv` library isn't.
I think we should modify one of them to make is consistent.

Reply:
I will use `$(LIBRABBITMQ_LIBRARY)` rather than `$(LIBS)`.

Reply:
I build this PR on de10-nano and there is a linking error,  `could not read symbols: Bad value`. It seems to need to recompile RabbitMQ with `-fPIC`, as refers to https://github.com/rust-lang/cargo/issues/713.

The current workaround by define `$(LIBTUV_LIBRARY_LINK)`
```
$(OUT)/test-%: $(OUT)/test-%.o $(OBJS) $(LIBTUV_LIBRARY)
     $(VECHO) "  LD\t$@\n"
     $(Q)$(CC) -o $@ $^ $(LDFLAGS) $(LIBTUV_LIBRARY_LINK)
```
```
$(LIBTUV_LIBRARY_LINK) = -Wl,-Bdynamic -L$(LIBRABBITMQ_PATH)/build/librabbitmq/ -llibrabbitmq
```
Any suggestion?

Reply:
The `libtuv` is compiled with the flag `-fPIC`, too.

Reply:
Do you mean we need to fork RabbitMQ C client into DLTcollab in order to add `-fPIC` flag and maintain   it?